js/tuxexchange.js

```diff
@@ -0,0 +1,533 @@
+        for (let i = 0; i < deposits.length; i++) {
+            const deposit = deposits[i];
+            if (deposit.txid !== undefined) {
+                validDeposits.push (deposit);
+            }
+        }
+        const result = this.parseTransactions (validDeposits, code, since, limit, { 'type': 'deposit' });
+        return result;
+    }
+
+    async fetchWithdrawals (code = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const withdrawals = await this.privatePost (this.extend ({ 'method': 'getmywithdrawhistory' }, params));
+        // Some withdrawals seem to have no data associated
+        const validWithdrawals = [];
+        for (let i = 0; i < withdrawals.length; i++) {
+            const withdrawal = withdrawals[i];
+            if (withdrawal.txid !== undefined) {
+                validWithdrawals.push (withdrawal);
+            }
+        }
+        const result = this.parseTransactions (validWithdrawals, code, since, limit, { 'type': 'withdrawal' });
+        return result;
+    }
+
+    async withdraw (code, amount, address, tag = undefined, params = {}) {
+        if (code === undefined || amount === undefined || address === undefined) {
+            throw new ArgumentsRequired (this.id + ' withdraw () requires a code, amount and address argument');
+        }
+        if (address.indexOf ('0x') === 0) {
+            address = address.substr (2);
+        }
+        this.checkAddress (address);
+        const withdrawRequest = {
+            'method': 'withdraw',
+            'coin': this.currencyId (code),
+            'address': address,
+            'amount': amount,
+        };
+        const result = await this.privatePost (this.extend (withdrawRequest, params));
+        return { 'info': result };
+    }
+
+    nonce () {
+        // The tuxexchange api actually ignores nonce in practice (even though it is documented)
+        return this.milliseconds ();
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = this.urls['api'];
+        const query = this.omit (params, this.extractParams (path));
+        let urlQueryParams = this.urlencode (query);
+        if (api === 'private') {
+            if (!headers) {
+                headers = {};
+            }
+            const nonce = this.nonce ();
+            urlQueryParams += '&nonce=' + nonce;
+            // Encoding the query params is non-standard and poorly documented but somehow correct
+            headers = this.extend (headers, {
+                'Key': this.apiKey,
+                'Sign': this.hmac (urlQueryParams, this.secret, 'sha512', 'hex'),
+            });
+        }
+        if (Object.keys (query).length) {
+            url += '?' + urlQueryParams;
+        }
+        return { 'url': url, 'method': method, 'body': body, 'headers': headers };
+    }
+
+    getIdsFromSymbol (symbol) {
```

Comment:
This method has to be dropped, because we already have `baseId` and `quoteId` in unified market structures.